runtime/local/git/git.go

```diff
@@ -1,157 +1,79 @@
-	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/teris-io/shortid"
 )
 
 type Gitter interface {
-	Clone(repo string) error
-	FetchAll(repo string) error
 	Checkout(repo, branchOrCommit string) error
-	RepoDir(repo string) string
+	RepoDir() string
 }
 
-type libGitter struct {
+type binaryGitter struct {
 	folder string
 }
 
-func (g libGitter) Clone(repo string) error {
-	fold := filepath.Join(g.folder, dirifyRepo(repo))
-	exists, err := pathExists(fold)
-	if err != nil {
-		return err
-	}
-	if exists {
-		return nil
-	}
-	_, err = git.PlainClone(fold, false, &git.CloneOptions{
-		URL:      repo,
-		Progress: os.Stdout,
-	})
-	return err
-}
-
-func (g libGitter) FetchAll(repo string) error {
-	repos, err := git.PlainOpen(filepath.Join(g.folder, dirifyRepo(repo)))
-	if err != nil {
-		return err
-	}
-	remotes, err := repos.Remotes()
-	if err != nil {
-		return err
-	}
-
-	err = remotes[0].Fetch(&git.FetchOptions{
-		RefSpecs: []config.RefSpec{"refs/*:refs/*", "HEAD:refs/heads/HEAD"},
-		Progress: os.Stdout,
-		Depth:    1,
-	})
-	if err != nil && err != git.NoErrAlreadyUpToDate {
-		return err
-	}
-	return nil
-}
-
-func (g libGitter) Checkout(repo, branchOrCommit string) error {
+func (g *binaryGitter) Checkout(repo, branchOrCommit string) error {
 	if branchOrCommit == "latest" {
 		branchOrCommit = "master"
 	}
-	repos, err := git.PlainOpen(filepath.Join(g.folder, dirifyRepo(repo)))
-	if err != nil {
-		return err
-	}
-	worktree, err := repos.Worktree()
-	if err != nil {
-		return err
-	}
+	// @todo if it's a commit it must not be checked out all the time
+	repoFolder := strings.ReplaceAll(strings.ReplaceAll(repo, "/", "-"), "https://", "")
+	g.folder = filepath.Join(os.TempDir(),
+		repoFolder+"-"+shortid.MustGenerate())
 
-	if plumbing.IsHash(branchOrCommit) {
-		return worktree.Checkout(&git.CheckoutOptions{
-			Hash:  plumbing.NewHash(branchOrCommit),
-			Force: true,
-		})
+	url := fmt.Sprintf("%v/archive/%v.zip", repo, branchOrCommit)
+	if !strings.HasPrefix(url, "https://") {
+		url = "https://" + url
 	}
-
-	return worktree.Checkout(&git.CheckoutOptions{
-		Branch: plumbing.NewBranchReferenceName(branchOrCommit),
-		Force:  true,
-	})
-}
-
-func (g libGitter) RepoDir(repo string) string {
-	return filepath.Join(g.folder, dirifyRepo(repo))
-}
-
-type binaryGitter struct {
-	folder string
-}
-
-func (g binaryGitter) Clone(repo string) error {
-	fold := filepath.Join(g.folder, dirifyRepo(repo), ".git")
-	exists, err := pathExists(fold)
+	resp, err := http.Get(url)
 	if err != nil {
-		return err
-	}
-	if exists {
-		return nil
+		return fmt.Errorf("Can't get zip: %v", err)
 	}
-	fold = filepath.Join(g.folder, dirifyRepo(repo))
-	cmd := exec.Command("git", "clone", repo, ".")
 
-	err = os.MkdirAll(fold, 0777)
-	if err != nil {
-		return err
-	}
-	cmd.Dir = fold
-	_, err = cmd.Output()
-	if err != nil {
-		return err
-	}
-	return err
-}
+	defer resp.Body.Close()
+	// Github returns 404 for tar.gz files...
```

Comment:
wat

Reply:
I can actualy remove this now as zip files are not throwing 404s....